src/wallet/wallet.cpp

```diff
@@ -2274,13 +2252,13 @@ bool CWalletTx::InMempool() const
     return fInMempool;
 }
 
-bool CWalletTx::IsTrusted(interfaces::Chain::Lock& locked_chain) const
+bool CWalletTx::IsTrusted() const
 {
     // Quick answer in most cases
-    if (!locked_chain.checkFinalTx(*tx)) {
+    if (!pwallet->chain().checkFinalTx(*tx)) {
```

Comment:
Instead of adding an `interfaces::Chain::checkFinalTx` method, it might be better to call [`IsFinalTx`](https://github.com/bitcoin/bitcoin/blob/50591f6ec61b802cf4193cdbefcc85ad75716e8d/src/consensus/tx_verify.cpp#L17-L28) directly with the wallet's height and time, avoiding going through [`CheckFinalTx`](https://github.com/bitcoin/bitcoin/blob/50591f6ec61b802cf4193cdbefcc85ad75716e8d/src/validation.cpp#L189-L219).

This could perform better since it wouldn't require locking cs_main, and could make wallet calls return more internally consistent information when the wallet is catching up to the chain.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/16426#discussion_r342719832

> Instead of adding an `interfaces::Chain::checkFinalTx` method, it might be better to call [`IsFinalTx`](https://github.com/bitcoin/bitcoin/blob/50591f6ec61b802cf4193cdbefcc85ad75716e8d/src/consensus/tx_verify.cpp#L17-L28)

#17443 is a step in this direction (https://github.com/bitcoin/bitcoin/pull/17443#pullrequestreview-332692782)